doc/dynamic-environments/configuration.mkd

```diff
@@ -200,12 +200,92 @@ sources:
 
 ### deploy
 
-The `deploy` setting is a new top level setting for controlling how r10k deploys
+The `deploy` setting is a top level setting for controlling how r10k deploys
 behave. At this point only new settings are included under this setting, but in
 the long term the current top level deploy settings will be moved under
 `deploy`.
 
-#### write_lock
+#### purge\_levels
+
+The `purge_levels` setting controls how aggressively r10k will purge unmanaged
+content during a deployment. Given value must be a list of strings indicating at
+what levels unmanaged content should be purged. The valid string options for the
+list are 'deployment', 'environment', and 'puppetfile'.
+
+```yaml
+---
+deploy:
+  purge_levels: [ 'deployment', 'environment', 'puppetfile' ]
+```
+
+This setting currently only impacts the "deploy environment" action.
+
+The default value is `['deployment', 'puppetfile']` to maintain parity with
+existing behavior before this setting was added.
+
+The effect of enabling the various purge levels is as follows:
+
+##### deployment
+
+After each deploy, in the configured basedir, r10k will recursively remove any
+content found which is not managed by one of the sources declared in the r10k.yaml
+configuration. Note that disabling this level of purging may cause the number of
+deployed environments to grow without bound; deleting branches from a control
+repo would no longer cause the matching environment to be purged.
+
+##### environment
+
+After a given environment is deployed, r10k will recursively remove any content
+found which is neither committed to the control repo branch that maps to that
+environment, nor declared in a Puppetfile committed to that branch.
+
+Enabling this purge level will cause r10k to load and parse the Puppetfile for
+the environment even without the `--puppetfile` flag being set. However,
+Puppetfile content will still only be deployed if the environment is new or
+the `--puppetfile` flag is set. Additionally, no environment-level content
+will be purged if a Puppetfile content deploy was triggered but an error was
+encountered.
+
+Note that the .r10k-deploy.json file is exempt from this purging.
+
+##### puppetfile
+
+After Puppetfile content for a given environment is deployed, r10k will
+recursively remove any content found in a directory managed by the Puppetfile
+which is not also declared in that Puppetfile. Directories considered to be
+managed by a Puppetfile include the default moduledir (typically "modules")
```

Comment:
I think this should be "the specified moduledir (defaults to "modules" if not specified)"  Current wording is not clear to me.